Userland/Libraries/LibCore/MimeData.cpp

```diff
@@ -72,11 +72,49 @@ String guess_mime_type_based_on_filename(const StringView& path)
         return "text/markdown";
     if (path.ends_with(".html", CaseSensitivity::CaseInsensitive) || path.ends_with(".htm", CaseSensitivity::CaseInsensitive))
         return "text/html";
+    if (path.ends_with(".js", CaseSensitivity::CaseInsensitive))
+        return "application/javascript";
+    if (path.ends_with(".json", CaseSensitivity::CaseInsensitive))
+        return "application/json";
+    if (path.ends_with(".md", CaseSensitivity::CaseInsensitive))
+        return "text/markdown";
     if (path.ends_with("/", CaseSensitivity::CaseInsensitive))
         return "text/html";
     if (path.ends_with(".csv", CaseSensitivity::CaseInsensitive))
         return "text/csv";
     return "text/plain";
 }
 
+#define ENUMERATE_HEADER_CONTENTS(V)                                                                   \
+    V(bmp, "image/bmp", 2, 'B', 'M')                                                                   \
+    V(bzip2, "application/x-bzip2", 3, 'B', 'Z', 'h')                                                  \
+    V(elf, "extra/elf", 4, 0x7F, 'E', 'L', 'F')                                                        \
+    V(gif_87, "image/gif", 6, 'G', 'I', 'F', '8', '7', 'a')                                            \
+    V(gif_89, "image/gif", 6, 'G', 'I', 'F', '8', '9', 'a')                                            \
+    V(gzip, "extra/gzip", 2, 0x1F, 0x8B)                                                               \
+    V(jpeg, "image/jpeg", 4, 0xFF, 0xD8, 0xFF, 0xDB)                                                   \
+    V(jpeg_huh, "image/jpeg", 4, 0xFF, 0xD8, 0xFF, 0xEE)                                               \
+    V(jpeg_jfif, "image/jpeg", 12, 0xFF, 0xD8, 0xFF, 0xE0, 0x00, 0x10, 'J', 'F', 'I', 'F', 0x00, 0x01) \
+    V(pbm, "image/x-portable-bitmap", 3, 0x50, 0x31, 0x0A)                                             \
+    V(pgm, "image/x-portable-graymap", 3, 0x50, 0x32, 0x0A)                                            \
+    V(png, "image/png", 8, 0x89, 'P', 'N', 'G', 0x0D, 0x0A, 0x1A, 0x0A)                                \
+    V(ppm, "image/x-portable-pixmap", 3, 0x50, 0x33, 0x0A)                                             \
+    V(shell, "text/x-shellscript", 10, '#', '!', '/', 'b', 'i', 'n', '/', 's', 'h', '\n')              \
+    V(tiff, "image/tiff", 4, 'I', 'I', '*', 0x00)                                                      \
+    V(tiff_bigendian, "image/tiff", 4, 'M', 'M', 0x00, '*')
+
+#define V(var_name, mime_type, pattern_size, ...) \
+    static const u8 var_name[pattern_size] = { __VA_ARGS__ };
```

Comment:
Any reason to not just construct the `ReadonlyBytes` here (rather than in each `guess_mime_type_based_on_sniffed_bytes` call)? Even has `constexpr` support :)

Reply:
I had a lot of trouble trying to understand the code in `Span.h`, so I couldn't figure out a way to construct it there without `no matching constructor` errors or weird hacky solutions.